src/realm/sort_descriptor.cpp

```diff
@@ -391,6 +402,47 @@ void FilterDescriptor::execute(const Table& table, KeyValues& key_values, const
     key_values = std::move(filtered);
 }
 
+template <typename T, typename U>
+inline int compare(const T& i, const T& j, const U& col)
+{
+    const auto id_i = i.cached_string_id;
```

Comment:
This code is massively overusing `auto` and is difficult to read as a result. The actual type of the variables are frequently not clear from context and also aren't things which would be unwieldy to write out.

Reply:
Fixed and made more clear + added more comments.